### PR TITLE
Use mock.patch.object to avoid protected access errors.

### PR DESCRIPTION
**Pull request by jameshilliard**

### CODE REVIEW

File: test/framers/test_tbc_transaction.py [hunk 1]
```diff
@@ -90,7 +91,9 @@ def test_calculate_exception_length(self):
             )
 
     @mock.patch("pymodbus.transaction.time")
-    def test_execute(self, mock_time):
+    @mock.patch.object(SyncModbusTransactionManager, "_recv")
```
Comment (janiversen) [inline test/framers/test_tbc_transaction.py:94]: This is wrong, it change the scope from class to local.

File: test/framers/test_tbc_transaction.py [hunk 2]
```diff
@@ -353,19 +353,18 @@ def callback(data):
         # for name in ("transaction_id", "protocol_id", "slave_id"):
         #     assert getattr(expected, name) == getattr(actual, name)
 
-    def test_tcp_framer_packet(self):
+    @mock.patch.object(ModbusRequest, "encode")
```
Comment (janiversen) [inline test/framers/test_tbc_transaction.py:356]: Seems a number of the patch objects could be done at class level.

**Review by janiversen — CHANGES_REQUESTED** [review]
A few comments, but the most important one:

You already submitted #2251 to solve this problem, seems you did not complete your work. Having 2 PR solving the same problem in the same file is over the top, and wasting my time, since I have to review them.

Please do not submit more PR to test_transaction.py unless they contain more than cosmetic changes, meaning real code change.

**Comment by jameshilliard** [comment]
> Having 2 PR solving the same problem in the same file is over the top, and wasting my time, since I have to review them.

Um, it's a different file right? Path for this PR looks to be `test/framers/test_tbc_transaction.py` while #2251 is for `test/test_transaction.py`. So `test/framers/test_tbc_transaction.py` is rather similar to `test/test_transaction.py`, are we supposed to have both of these tests or just one?

**Comment by janiversen** [comment]
If you had a look at the files you would see it’s the same file, just copied to prepare for major changes.

**Comment by jameshilliard** [comment]
> If you had a look at the files you would see it’s the same file,

Yeah...I missed that somehow, had assumed it wasn't an exact duplicate so had applied the same fixes from the other one.

> just copied to prepare for major changes.

Um, so I should make sure changes are duplicated between the files?

**Comment by janiversen** [comment]
No you should not touch a file marked "to be changed" unless you want to change it as intended.

**Comment by jameshilliard** [comment]
> change it as intended.

How is it intended to be changed?

**Comment by janiversen** [comment]
Read the issues, it will be integrated in framers.

**Pull request closed without merging**